xdsl/dialects/scf.py

```diff
@@ -169,44 +169,54 @@ def get(
         return For(lb, ub, step, iter_args, body)
 
     def verify_(self):
-        if (len(self.iter_args) + 1) != len(self.body.block.args):
+        # body block verification
+        if len(self.body.block.args) == 0:
             raise VerifyException(
-                f"Wrong number of block arguments, expected {len(self.iter_args)+1}, got "
-                f"{len(self.body.block.args)}. The body must have the induction "
-                f"variable and loop-carried variables as arguments."
+                "Body block must have induction var as first block arg"
             )
-        if self.body.block.args and (iter_var := self.body.block.args[0]):
-            if not isinstance(iter_var.type, IndexType):
+
+        indvar, *block_iter_args = self.body.block.args
+        block_iter_args_num = len(block_iter_args)
+        iter_args = self.iter_args
+        iter_args_num = len(self.iter_args)
+
+        for i, opnd in enumerate(self.operands[:3]):
```

Comment:
the :3 is a little scary, maybe something like this?
```suggestion
        for opnd in (self.lb, self.ub, self.step):
```
i also seems not to be used

Reply:
That's there because the current `signlessInteger` constrained type doesn't catch the case when all signed integers are used, which is a bug and I'm trying to resolve it in #1727.

Not sure what the best course of action is here.

Reply:
Move the test from #1727 that should fail here?

Reply:
maybe an expectedly failed test?

Reply:
I made a mistake here; This actually used and exercised by the currently added test cases.